src/Compilers/Core/Rebuild/BuildConstructor.cs

```diff
@@ -24,23 +26,20 @@
 
 namespace BuildValidator
 {
-    /// <summary>
-    /// An abstraction for building from a MetadataReaderProvider
-    /// </summary>
-    internal class BuildConstructor
+    public class BuildConstructor
```

Comment:
>public [](start = 4, length = 6)

Should the types in this assembly be kept as `internal` with `InternalsVisibleTo()` access added for `BuildValidator`?

Reply:
We do intend for external users to be able to consume this library. I want to punt on the decision of which members to reduce to internal accessibility until we get closer to the point where the public API is ready for review.

Reply:
> We do intend for external users to be able to consume this library.

If these types will be public, we should also review the namespace names when the public API is reviewed.

---
In reply to: [589681272](https://github.com/dotnet/roslyn/pull/51709#discussion_r589681272) [](ancestors = 589681272)